src/trap.js

```diff
@@ -390,6 +434,16 @@ class Trap {
   generateNewStreamId() {
     this._metadata.generateNewStreamId();
   }
+
+  /**
+   * Returns the collected events that have not been flushed yet.
+   * Only works if inMemor
+   *
+   * @returns {undefined|Array<Array>}
+   */
+  getCollectedEvents() {
```

Comment:
IMO this function name implies that this method won't modify the current state, i.e. it won't flush the storage.  Please use a different name if it changes (i.e. clears) the buffer.